src/Uno.UI/UI/Xaml/UIElement.cs

```diff
@@ -844,7 +844,30 @@ private static void InnerUpdateLayout(UIElement root)
 
 		internal void ApplyClip()
 		{
-#if __CROSSRUNTIME__
+#if __SKIA__
+			// On Skia specifically, we separate the two types of clipping.
+			// First, from Clip DP (handled in this code path)
+			// That clipping propagates to Visual.Clip through ApplyNativeClip.
+			// Second is clipping calculated from FrameworkElement.GetClipRect during arrange.
+			// That clipping propagates to ViewBox during arrange.
+			var clip = Clip;
+			if (clip is null)
+			{
+				ApplyNativeClip(Rect.Empty);
+				OnViewportUpdated();
+			}
+			else
+			{
+				var rect = clip.Rect;
+				if (clip.Transform is { } clipTransform)
+				{
+					rect = clipTransform.TransformBounds(rect);
```

Comment:
On skia, we should instead map this to the `CompositionClip.TransformMatrix`.

Reply:
This is the `UIElement.Clip` property. It's not a composition clip. Or am I misunderstanding something?

Reply:
On ski the `ApplyNativeClip` will actually only set the `Visual.Clip` property which is of type `CompositionClip` and itself which has a `TransformMatrix` property. SO the idea is that instead of transforming the `rect` here, I would just push the transform to the `CompositionClip.TransformMatrix`. This will later allow us to animate that transform "independently".